crates/pop-parachains/src/templates.rs

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: GPL-3.0
+#[derive(Debug, Clone, PartialEq)]
+pub struct Config {
+	pub symbol: String,
+	pub decimals: u8,
+	pub initial_endowment: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Template {
+	// Pop
+	Base,
+	// Parity
+	ParityContracts,
+	ParityFPT,
+}
+impl Template {
+	pub fn is_provider_correct(&self, provider: &Provider) -> bool {
+		match provider {
+			Provider::Pop => self == &Template::Base,
+			Provider::Parity => self == &Template::ParityContracts || self == &Template::ParityFPT,
+		}
+	}
+	pub fn from(provider_name: &str) -> Self {
+		match provider_name {
+			"base" => Template::Base,
+			"cpt" => Template::ParityContracts,
+			"fpt" => Template::ParityFPT,
+			_ => Template::Base,
+		}
+	}
+	pub fn repository_url(&self) -> &str {
+		match &self {
+			Template::Base => "r0gue-io/base-parachain",
+			Template::ParityContracts => "paritytech/substrate-contracts-node",
+			Template::ParityFPT => "paritytech/frontier-parachain-template",
+		}
+	}
+}
```

Comment:
Since we are making this code to be future compatible with different providers and templates, I suggest a more refined approach rather than relying on manual matches. Look at `strum` or string enums an already imported library which I have used in the past to make string associations to enum variants. 

One enum variant can be made to match to a `url`, a `provider` and ofc any other name as desired. This functionality need not  be handwritten.

Reply:
Good point. Same as above, solved after Frank's refactor